python-checks/src/main/java/org/sonar/python/checks/tests/TestShouldBeSkippedExplicitlyCheck.java

```diff
@@ -0,0 +1,128 @@
+  private static final Tree.Kind[] literalsKind = {Tree.Kind.STRING_LITERAL, Tree.Kind.NUMERIC_LITERAL, Tree.Kind.LIST_LITERAL,
+    Tree.Kind.BOOLEAN_LITERAL_PATTERN, Tree.Kind.NUMERIC_LITERAL_PATTERN, Tree.Kind.NONE_LITERAL_PATTERN, Tree.Kind.STRING_LITERAL_PATTERN,
+    Tree.Kind.SET_LITERAL, Tree.Kind.DICTIONARY_LITERAL};
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.FUNCDEF, ctx -> {
+      FunctionDef functionDef = (FunctionDef) ctx.syntaxNode();
+
+      if (!functionDef.name().name().startsWith("test")) {
+        return;
+      }
+
+      ReturnStatement returnStatement = getReturnStatementInIfFirstStatement(functionDef);
+      if (returnStatement == null) {
+        return;
+      }
+
+      // check there is an assert like statement in the function
+      if (isOrContainAssertStatement(functionDef)) {
+        ctx.addIssue(returnStatement, MESSAGE);
+      }
+    });
+  }
+
+  private static ReturnStatement getReturnStatementInIfFirstStatement(FunctionDef functionDef) {
+    // check first non assignment statement is an if with a return
+    for (Statement statement : functionDef.body().statements()) {
+      // skip literal assignment expression
+      if (statement.is(Tree.Kind.ASSIGNMENT_STMT) && ((AssignmentStatement) statement).assignedValue().is(literalsKind)) {
+        continue;
+      }
+
+      if (statement.is(Tree.Kind.IF_STMT)) {
+        IfStatement ifStatement = (IfStatement) statement;
+        List<Statement> statements = ifStatement.body().statements();
+        if (statements.get(0).is(Tree.Kind.RETURN_STMT)) {
+          return (ReturnStatement) statements.get(0);
+        }
+      }
+
+      return null;
+    }
+    return null;
+  }
+
+  private static boolean isOrContainAssertStatement(Tree node) {
+    AssertOrCallAssertVisitor assertOrCallAssertVisitor = new AssertOrCallAssertVisitor();
+    node.accept(assertOrCallAssertVisitor);
+    return assertOrCallAssertVisitor.hasACallToAssert || assertOrCallAssertVisitor.hasANativeAssert;
+  }
+
+  static class AssertOrCallAssertVisitor extends BaseTreeVisitor {
+    boolean hasANativeAssert = false;
+    boolean hasACallToAssert = false;
+
+    @Override
+    public void visitAssertStatement(AssertStatement pyAssertStatementTree) {
+      hasANativeAssert = true;
+      super.visitAssertStatement(pyAssertStatementTree);
+    }
+
+    @Override
+    public void visitQualifiedExpression(QualifiedExpression pyQualifiedExpressionTree) {
+      if (tryGetName(pyQualifiedExpressionTree.qualifier()).equals("self")
+        && supportedAssertMethods.contains(pyQualifiedExpressionTree.name().name())) {
+        hasACallToAssert = true;
+      }
+      super.visitQualifiedExpression(pyQualifiedExpressionTree);
+    }
+
+    private static String tryGetName(Expression expression) {
+      if (expression.is(Tree.Kind.NAME)) {
+        return ((Name) expression).name();
+      }
+      return "";
+    }
```

Comment:
In order not to return "", you could use `Optional.of()` over `expression` and then filter it over the name and check whether it is in the set with `isPresent()`. But that's merely a suggestion

Reply:
I like the approach, will implement this.